lighthouse-core/test/chromium-web-tests/test-page-devtools.sh

```diff
@@ -0,0 +1,108 @@
+
+set -euo pipefail
+
+# Get newest folder
+latest_content_shell_dir=$(ls -t "$LH_ROOT/.tmp/chromium-web-tests/content-shells/" | head -n1)
+latest_content_shell="$LH_ROOT/.tmp/chromium-web-tests/content-shells/$latest_content_shell_dir"
+
+# Run a very basic server on port 8000. Only thing we need is:
+#   - /devtools -> the layout tests for devtools frontend
+#   - /inspector-sources -> the inspector resources from the content shell
+#   - CORS (Access-Control-Allow-Origin header)
+
+# Setup inspector-sources.
+yarn devtools "$DEVTOOLS_PATH"
+cd "$DEVTOOLS_PATH"
+autoninja -C out/Default # Build devtools resources.
+cd -
+ln -s "$DEVTOOLS_PATH/out/Default/resources/inspector" "$DEVTOOLS_PATH/test/webtests/http/tests/inspector-sources"
+
+# Add test to run lighthouse in DevTools and print LHR.
+echo "
+(async function() {
+  await TestRunner.navigatePromise('$1');
+
+  await TestRunner.loadModule('lighthouse_test_runner');
+  await TestRunner.showPanel('lighthouse');
+
+  LighthouseTestRunner.getRunButton().click();
+  const {lhr} = await LighthouseTestRunner.waitForResults();
+  TestRunner.addResult(JSON.stringify(lhr));
+
+  TestRunner.completeTest();
+})();
+" > "$DEVTOOLS_PATH/test/webtests/http/tests/devtools/lighthouse/lighthouse-run-dt.js"
+
+# Kill background jobs and remove temporary files when script ends.
+cleanup() {
+  rm "$DEVTOOLS_PATH/test/webtests/http/tests/inspector-sources"
+  rm "$DEVTOOLS_PATH/test/webtests/http/tests/devtools/lighthouse/lighthouse-run-dt.js"
+  kill ${SERVER_PID}
+}
+trap 'cleanup' EXIT
+
+# Serve from devtools frontend webtests folder.
+(npx http-server@0.12.3 "$DEVTOOLS_PATH/test/webtests/http/tests" -p 8000 --cors > /dev/null 2>&1) &
+SERVER_PID=$!
+
+echo "Waiting for server"
+health_check_url='http://localhost:8000/inspector-sources/integration_test_runner.html?experiments=true&test=http://127.0.0.1:8000/devtools/lighthouse/lighthouse-view-trace-run.js'
+until $(curl --output /dev/null --silent --head --fail $health_check_url); do
+  printf '.'
+  sleep 1
+done
+echo "Server is up"
+
+# webtests sometimes error if results are already present.
+rm -rf "$latest_content_shell/out/Release/layout-test-results"
+
+# Add typ to python path. The regular method assumes there is a Chromium checkout.
+export PYTHONPATH="${PYTHONPATH:-}:$BLINK_TOOLS_PATH/latest/third_party/typ"
+
+# Don't quit if the python command fails.
+set +e
+
+python \
+  "$BLINK_TOOLS_PATH/latest/third_party/blink/tools/run_web_tests.py" \
+  --layout-tests-directory="$DEVTOOLS_PATH/test/webtests" \
+  --build-directory="$latest_content_shell/out" \
+  --no-show-results \
+  --time-out-ms=60000 \
+  http/tests/devtools/lighthouse/lighthouse-run-dt.js
+
+set -e
+
+rm -rf "$LH_ROOT/.tmp/layout-test-results"
+
+# Copy results to latest-run folder.
+# Sometimes there will be extra output before the line with LHR. To get around this, only copy the last line with content.
+awk '/./{line=$0} END{print line}' \
```

Comment:
do you have an example of this happening? based on the comment alone my immediate thought is "should we just grep for the LHR using lighthouseVersion or fetchTime or something instead?"

if there's a good reason for the awk approach example might help illustrate that :)

Reply:
Running the script on `https://google.com` had some extra output. There is not a specific reason for the awk approach, that was just the first solution I found that worked. The goal was to fetch the last line of output, but a grep solution would be more robust.

Reply:
grep is also a smidge more intuitive from my perspective but I never really got deep into awk to ever feel like any of its uses are very simple either so I might be biased :)